### PR TITLE
feat(bsn): add proof-of-concept transport token utils

### PR DESCRIPTION
**Pull request by JorisHeadease**
This PR just includes the basic functions used by the "Knooppunten". It also contains the functions for the NVI. The latter should be removed, but we mainly wanted to prove the principle in a single test.

  **Knooppunt functions:**
  - `CreateTransportToken(bsn int, audience string)` - BSN → unique transport token
  - `BSNFromTransportToken(token string)` - Extract BSN from token

  **NVI functions (to be removed later):**
  - `TransportTokenToPseudonym(token string)` - Token → consistent pseudonym
  - `PseudonymToTransportToken(pseudonym string, audience string)` - Pseudonym → new org-specific token

  **Testing:**
  - Full test suite validating the complete workflow
  - Some edge cases

  **Note:** Uses simple XOR encryption (proof-of-concept only)

### CODE REVIEW

File: lib/bsnutil/util.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+// CreateTransportToken creates a transport token from BSN and audience using simple XOR transformation.
+// Each transport token is unique (includes random nonce) but contains the same encrypted BSN.
+// This ensures transport tokens cannot be tracked while the NVI can always generate the same pseudonym.
+//
+// Parameters:
+//   - bsn: The Dutch social security number (BSN)
+//   - audience: The identifier for the organization/audience receiving the token
+//
+// Returns a transport token in format: "token-{audience}-{transformedBSN}-{nonce}"
+func CreateTransportToken(bsn int, audience string) (string, error) {
```
Comment (reinkrul) [inline lib/bsnutil/util.go:33]: I'd make BSN a `string`, as it fixed length, at some systems may use prefix-zeroes

File: lib/bsnutil/util.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	key := generateSimpleKey(pseudonymHolder)
+	originalBSN := simpleXOR(transformedBSN, key)
+
+	// Create new token with the target audience
+	return CreateTransportToken(originalBSN, audience)
+}
+
+// validateBSN checks if the BSN is within valid range for Dutch social security numbers.
+func validateBSN(bsn int) error {
+	if bsn < minBSN || bsn > maxBSN {
```
Comment (reinkrul) [inline lib/bsnutil/util.go:160]: I don't think we need to validate this here, since we can work with a simple plaintext, which can be whatever data (no need to check business rules here)

File: lib/bsnutil/util_test.go [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+func TestCompleteWorkflow(t *testing.T) {
+	// Test the complete federated health workflow from the architecture diagram
+	bsn := 123456789
+
+	// LOCALIZATION: Knooppunt A registers DocumentReference
+	// Step 1: Knooppunt A creates transport token with audience="nvi"
+	tokenFromA, err := CreateTransportToken(bsn, "nvi")
+	if err != nil {
```
Comment (reinkrul) [inline lib/bsnutil/util_test.go:189]: can you use `require.NoError()` for shorter tests?

**Review by reinkrul — APPROVED** [review]
